### PR TITLE
4605-button

### PR DESCRIPTION
**Pull request by adamshcao**
image

### CODE REVIEW

File: components/src/button/Button.tsx [hunk 1]
```diff
@@ -36,52 +36,39 @@ interface ButtonProps {
...
-    focus = "focus:ring-2 focus:ring-warning-main focus:ring-offset-2 "
-  }
+        buttonType,
+        onLightBg,
+        backgroundColor,
+        label,
+        ...props
+   }: ButtonProps) => {
+    let base,hover,active,focus
+    if (buttonType === "secondary" && onLightBg) {
```
Comment (snony14) [inline components/src/button/Button.tsx:46]: Is there an easier way to define all of these classes, many of the classes repeat.

File: components/src/button/Button.tsx [hunk 2]
```diff
@@ -36,52 +36,39 @@ interface ButtonProps {
...
+        base = "font-roboto text-[11px] rounded-full w-auto h-21px px-4 bg-transparent text-primary-dark outline outline-1 outline-offset--1 outline-primary-dark ";
+        hover = "hover:bg-primary-dark hover:text-white hover:outline-none"
+        active = "active:bg-transparent active:text-primary-dark active:outline active:outline-1 active:outline-offset--1 active:ring-offset-0 active:outline-primary-dark active:ring-0 "
+        focus = "focus:ring-2 focus:ring-warning-main focus:ring-offset-2 active:bg-transparent "
+    }else if (buttonType === "secondary" && !onLightBg) {
```
Comment (algreasley) [inline components/src/button/Button.tsx:51]: could you sort out prettier in IDEA please .. I am going to make it a condition for approving this PR 😛

File: components/src/button/Button.tsx (hunk 1 above)
Comment (algreasley) [reply components/src/button/Button.tsx:46]: yeah, does seem like a real ocean of classes .. I know that is, in general, "the tailwind way" but let's see if we can reduce the duplication ..

**Review by algreasley — CHANGES_REQUESTED** [review]
Our end state should be more like this, with controls to toggle the state of the button
I know that is not clear from the ACs so this is not a criticism of this PR

ui-kit-2022--button-with-controls

**Review by tammyk7 — APPROVED** [review]

**Pull request merged**